compiler/rustc_serialize/src/opaque.rs

```diff
@@ -300,13 +300,13 @@ impl<'a> serialize::Decoder for Decoder<'a> {
     #[inline]
     fn read_char(&mut self) -> Result<char, Self::Error> {
         let bits = self.read_u32()?;
-        Ok(::std::char::from_u32(bits).unwrap())
+        Ok(std::char::from_u32(bits).unwrap())
     }
 
     #[inline]
     fn read_str(&mut self) -> Result<Cow<'_, str>, Self::Error> {
         let len = self.read_usize()?;
-        let s = ::std::str::from_utf8(&self.data[self.position..self.position + len]).unwrap();
+        let s = std::str::from_utf8(&self.data[self.position..][..len]).unwrap();
```

Comment:
I personally prefer the previous version here

Reply:
You mean the two slices? I can change it back if you want.

Reply:
doesn't really matter, but yeah... doing `[start..start + len]` is preferable over `[start..][..len]` to me.